### PR TITLE
Don't use JAVA_HOME in PropertiesTest

### PR DESCRIPTION
**Pull request by gmethvin**

### CODE REVIEW

File: test/junit/scala/util/PropertiesTest.scala [hunk 1]
```diff
@@ -37,7 +37,8 @@ class PropertiesTest {
     assertEquals(Properties.envOrElse("_PropertiesTest_NOT_DEFINED", "test"), "test")
 
     var done = false
-    assertNotEquals(Properties.envOrElse("JAVA_HOME", { done = true; "bar" }), "bar")
+    val envName = System.getenv().keySet().iterator().next()
+    assertNotEquals(Properties.envOrElse(envName, { done = true; "bar" }), "bar")
```
Comment (som-snytt) [inline test/junit/scala/util/PropertiesTest.scala]: I'm so ignorant, I don't about environments where the map is empty. You could harden it here, or kick that down the road.

File: test/junit/scala/util/PropertiesTest.scala [hunk 2]
```diff
@@ -37,7 +37,8 @@ class PropertiesTest {
...
     assertTrue("Does not evaluate alt if not needed", done == false)
```
Comment (som-snytt) [inline test/junit/scala/util/PropertiesTest.scala]: i.e., `assertFalse(done)`. But nbd.

**Review by som-snytt — APPROVED** [review]
Thanks, I was going to do a quickie, then I arrived at the office and it was already done!

**Comment by SethTisue** [comment]
did this come up somewhere somehow?

**Comment by som-snytt** [comment]
@SethTisue I noticed it trying to get a clean build locally, on that PR where I touch a thousand tests, or is it ten thousand? I commented on the linked PR.

**Comment by som-snytt** [comment]
Thanks everyone.

**Pull request merged**